bolt-sidecar/src/crypto/bls.rs

```diff
@@ -36,7 +38,6 @@ pub trait SignableBLS {
 }
 
 /// A generic signing trait to generate BLS signatures.
-#[allow(dead_code)]
 pub trait SignerBLS {
     /// Sign the given data and return the signature.
     fn sign(&self, data: &[u8]) -> eyre::Result<BLSSig>;
```

Comment:
Since BLS signers should only sign hash tree roots, perhaps we can change the param (here and in `SignerBLSAsync` to `root: [u8; 32]`?

Reply:
yes makes sense done!